### ISSUE TITLE
Setting Columns with JSON? (QUESTION/ISSUE)

### ISSUE THREAD

**Issue by cgountanis** [wenzhixin/bootstrap-table#3895]
Is it possible to set the columns property with JSON? Example below shows setting the data with AJAX JSON data but can it be done for the columns as well?

```
 $table.bootstrapTable({
                filterControl: true,
                disableUnusedSelectOptions: true,
                data: data,
                height: getHeight(),
                search: true,
                columns: [{
                    field: 'selected',
                    checkbox: true,
                    align: 'center'
                },
```

I have tried with this JSON data and no columns show up.

```
[
{"field":"selected","checkbox":true,"align":"center","title":null,"formatter":null,"sortable":false,"filterControl":null,"filterControlPlaceholder":null,"filterStrictSearch":false},

{"field":null,"checkbox":false,"align":"center","title":"Actions","formatter":"actionsFormatter","sortable":false,"filterControl":null,"filterControlPlaceholder":null,"filterStrictSearch":false},

{"field":"id","checkbox":false,"align":"center","title":"ID","formatter":null,"sortable":true,"filterControl":"input","filterControlPlaceholder":"Search","filterStrictSearch":false},

{"field":"cabinet","checkbox":false,"align":"center","title":"Cabinet","formatter":null,"sortable":true,"filterControl":"input","filterControlPlaceholder":"Search","filterStrictSearch":false},

{"field":"drawer","checkbox":false,"align":"center","title":"Drawer","formatter":null,"sortable":true,"filterControl":"input","filterControlPlaceholder":"Search","filterStrictSearch":false}
]
```

**Comment by djhvscf** [comment]
Please provide a [jsFiddle](https://github.com/wenzhixin/bootstrap-table/blob/develop/CONTRIBUTING.md#bug-reports) to show your problem, thanks!

**Comment by cgountanis** [comment]
The example is given you can not provide an example as the JSON would have to come from a service that is internal to the application.

So in my example above I am showing how to set the columns with JS. The data above is a JSON variable. I would like to do the same thing with columns.

The question is more-less can you set the bootstrap table properties like "columns" with a variable? I have had no luck so far. I could maybe go as far as the whole object as JSON if that is easier.

**Comment by djhvscf** [comment]
You can use [myJson.con](http://myjson.com) in order to have a fake "serverside"

**Comment by cgountanis** [comment]
OK, think this should do to get the basic point across.

https://jsfiddle.net/ChrisGo/94vu6p0L/

**Comment by djhvscf** [comment]
See these examples:

http://jsfiddle.net/wenyi/e3nk137y/14/ from url
http://jsfiddle.net/wenyi/e3nk137y/13/ from json object

You will need to define the columns and then you can load the data from a url or a json object.

**Comment by djhvscf** [comment]
I just found an issue with this approach, let me show you:

This [line](https://github.com/wenzhixin/bootstrap-table/blob/develop/src/bootstrap-table.js#L177) should be:

```javascript
if (func !== null && typeof func === 'object') {
    return func
}
```

**Comment by cgountanis** [comment]
Is there a BETA CDN I can switch in the Fiddle to test quick?

**Comment by djhvscf** [comment]
At this moment we need to apply that change under development branch so you can test. 

Also you can try to modify that line on your end and let us know the results.

**Comment by cgountanis** [comment]
Confirmed it seems to be accepting the JSON and populating the data now.

![image](https://user-images.githubusercontent.com/14899817/43150846-8af01610-8f30-11e8-975d-d48d633271ae.png)

Since then though we have decided to use JSON for the DATA in the HTML and just dynamically create the THEAD part of the HTML to save some headaches.

Good fix though for the next release in case someone wants to dynamically specify data and columns via JSON within JS directly though!
